pytorch_lightning/plugins/precision/native_amp.py

```diff
@@ -109,23 +109,20 @@ def forward_context(self) -> Generator[None, None, None]:
             yield
 
     def state_dict(self) -> Dict[str, Any]:
+        """``NativeMixedPrecisionPlugin.on_save_checkpoint`` is deprecated in v1.6.
```

Comment:
the docstrings are in the wrong method. did you mean to implement on_save/load_checkpoint in this class, and then call super().on_save_checkpoint / super().on_load_checkpoint ?

Reply:
we made them empty [in Stateful PR discussion](https://github.com/PyTorchLightning/pytorch-lightning/pull/11638#discussion_r799107580), and here I moved the docstring to avoid deprecation warnings - open to suggestions

Reply:
ah okay

Reply:
I think this is a very confusing choice.

Why don't we set the base class empty method to do


```py
f"""``{self.__class__.__name__}.on_save_checkpoint``is deprecated...
...
"""
```